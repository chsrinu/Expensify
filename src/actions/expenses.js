import uuid from 'uuid';

export const AddExpense = ({
    description = 'testDescription',
    note = 'testNote',
    amount = 1,
    createdAt = 0
    } = {}
) => (
        {
            type: 'EXPENSE',
            expense : {
                id: uuid(),
                description: description,
                note: note,
                amount: amount,
                createdAt: createdAt
            }
        }
    )
export const editExpense = (id,updates) => (
    {
        type:'EDIT_EXPENSE',
        id:id,
        updates:updates
    }
)
export const removeExpense = ({id}) => {
    console.log(id);
    return {
        type: 'REMOVE_EXPENSE',
        id: id
    }
}