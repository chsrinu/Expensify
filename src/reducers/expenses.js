export const expensesReducer = (state=[],action) => {
    switch(action.type){
        case 'EXPENSE':
            return [...state,action.expense];
        case 'REMOVE_EXPENSE':
           // console.log('deleting expense',action.id);
            return (state.filter((item)=>{
                if(item.id!=action.id)
                    return item;
            }))
        case 'EDIT_EXPENSE':
            console.log(action.id, action.updates);
            return state.map((expense)=>{
                if(expense.id===action.id)
                    return {...expense,...action.updates}
                return expense;
            })
        default:
        return state;
    }
}