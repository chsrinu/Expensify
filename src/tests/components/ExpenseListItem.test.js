import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import ExpenseListItem from '../../components/ExpenseListItem';

test('ExpenseListItem testCase', () => {
    const wrapper = shallow(<ExpenseListItem item = {expenses[0]} />)
    expect(wrapper).toMatchSnapshot();
});