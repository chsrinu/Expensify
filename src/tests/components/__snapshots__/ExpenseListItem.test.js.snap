// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpenseListItem testCase 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/0"
  >
    <h3>
      Google
    </h3>
  </Link>
  <p>
    10000
    -
    0
  </p>
</div>
`;
